crates/kernel/testsuite/moot_suite.rs

```diff
@@ -144,7 +174,7 @@ fn test(path: &Path) {
     let f = BufReader::new(File::open(path).unwrap());
     let db = create_db();
 
-    let mut state = MootState::default();
+    let mut state = MootState::new(Arc::new(NoopClientSession::new()));
```

Comment:
so is this really what we'd want, one session per the whole test run? or would we want it per test... in most testing frameworks each test is hermetic and isolated... though that is already not the case here on account of them all sharing the same mutable core

Reply:
IMO it's crucial that there's some grouping of tests that run on the same mutable core. Otherwise, testing side-effects / mutations gets really inconvenient.

And so, I *think* aligning the session lifetime with the core lifetime is the sensible approach.